app/charts/shared/chart-data-filters.tsx

```diff
@@ -255,14 +269,63 @@ const DataFilterBaseDimension = ({
       id="dataFilterBaseDimension"
       label={label}
       options={allOptions}
-      optionGroups={optionGroups}
       value={value}
       tooltipText={tooltipText || undefined}
       onChange={onChange}
     />
   );
 };
 
+const DataFilterHierarchyDimension = ({
+  dimension,
+  value,
+  onChange,
+  hierarchy,
+}: {
+  dimension: Dimension;
+  value: string;
+  onChange: (e: { target: { value: string } }) => void;
+  hierarchy?: HierarchyValue[];
+}) => {
+  const noneLabel = t({
+    id: "controls.dimensionvalue.none",
+    message: `No Filter`,
+  });
+
+  const {
+    label,
+    isKeyDimension,
+    description: tooltipText,
+    values: dimensionValues,
+  } = dimension;
+  const options = React.useMemo(() => {
+    let opts = [] as { label: string; value: string; isNoneValue?: boolean }[];
+    if (hierarchy) {
+      opts = hierarchyToOptions(hierarchy);
+    } else {
+      opts = dimensionValues;
+    }
+    if (!isKeyDimension) {
+      opts.push({
+        value: FIELD_VALUE_NONE,
+        label: noneLabel,
+        isNoneValue: true,
+      });
+    }
+    return opts;
+  }, [hierarchy, isKeyDimension, dimensionValues, noneLabel]);
```

Comment:
Did not check this as I'm not aware of optional hierarchy in any of the datasets, but wouldn't this append the "None" option at the end of the array (I think it should be at top)?

Reply:
Ah yes ! I replaced push by unshift :)